### ISSUE TITLE
[MU4 Issue] 'Create custom palette' dialog should respond to enter key

### ISSUE THREAD

**Issue by avvvvve** [musescore/MuseScore#19976]
### Issue type

UI bug

### Bug description

After typing in a name in the 'Create custom palette' dialog, the enter key should create the custom palette. The user currently must click the button.

### Steps to reproduce

1. Open the Palettes panel
2. Click 'Add palettes'
3. Click 'Create custom palette'
4. Type in a name
5. Hit enter, see that nothing happens

### Screenshots/Screen recordings

<img width="304" alt="image" src="https://github.com/musescore/MuseScore/assets/20806406/328d88c1-9b4e-4cd7-96aa-198f62c6145f">

### MuseScore Version

MuseScore version (64-bit): 4.1.1-232071203, revision: github-musescore-musescore-e4d1ddf

### Regression

I don't know

### Operating system

OS: macOS 13.6, Arch.: x86_64

**Comment by bkunda** [comment]
I suspect this is because our keyboard navigation system requires the user to tab to focus the buttons. 
There are probably other instances where this behaviour is similar. 
Will need to review this one. 
@shoogle FYI

**Comment by avvvvve** [comment]
Yeah, currently you have to hit tab to focus on the text field and then press the down arrow key twice to get to the Create button.

I think it'd be an easy win for keyboard users to allow the enter key to submit single-field forms like this.
